keps/sig-architecture/2527-clarify-status-observations-vs-rbac/README.md

```diff
@@ -0,0 +1,697 @@
+to write values to status that represent allocations or acknowledged requests.
+Document that status fields are best when they represent observed state, but do
+not _require_ it.
+
+This has [already happened](https://github.com/kubernetes/enhancements/pull/2308/files#r567809465)
+at least once.
+
+#### Examples
+
+Some of these are variants of existing APIs and some are hypothetical.
+
+1) User creates a Service and sets `Service.spec.type = LoadBalancer`.  The
+cloud controller sets `Service.status.loadBalancer.ingress.ip` in response.  If
+the user sets `Service.spec.loadBalancerIP` to a specific value, the cloud
+controller will either successfully use that IP and and set it as before (ACK),
+or it will not set any value in `Service.status.loadBalancer.ingress` (NAK).
+
+2) Given a Pod, user patches `Pod.spec.containers[0].resources.requests[cpu]`
+to a new value.  Kubelet sees this as a request and, if viable, sets
+`Pod.status.containerStatuses[0].resources.requests[cpu]` to the same value.
+
+3) User creates a `Bucket`, setting `Bucket.spec.instance = artifacts-prod`.
+The bucket controller verifies that the namespace is allowed to use the
+artifacts-prod bucket and, if so, sets `Bucket.status.instance` to the same
+value.
+
+#### Tradeoffs
+
+Pro: Easy and compatible with existing uses.
+
+Con: Dilutes the meaning of status.
+
+### Option 2: Add a new top-level stanza to spec/status resources
+
+Keep and strengthen the idea that status fields must be from observation.
+Segregate controller-owned fields to a new stanza, parallel to `spec` and
+`status`, which can be RBAC'ed explicitly.  For the sake of this doc, let's
+call it `control`.
+
+To make this viable, we would need to add `control` as a "standard" subresource
+and apply similar rules to `spec` and `status` with regards to writes (can't
+write to `control` through the main resource, can't write to `spec` or `status`
+through the `control` subresource).  We would also need to add it to CRDs as an
+available subresource.
+
+#### Examples
+
+1) User creates a Service and sets `Service.spec.type = LoadBalancer`.  The
+cloud controller sets `Service.control.loadBalancer.ingress.ip` in response.  If
+the user sets `Service.spec.loadBalancerIP` to a specific value, the cloud
+controller will either successfully use that IP and and set it as before (ACK),
+or it will not set any value in `Service.control.loadBalancer.ingress` (NAK).
+
+2) Given a Pod, user patches `Pod.spec.containers[0].resources.requests[cpu]`
+to a new value.  Kubelet sees this as a request and, if viable, sets
+`Pod.control.containerStatuses[0].resources.requests[cpu]` to the same value.
+
+3) User creates a `Bucket`, setting `Bucket.spec.instance = artifacts-prod`.
+The bucket controller verifies that the namespace is allowed to use the
+artifacts-prod bucket and, if so, sets `Bucket.control.instance` to the same
+value.
+
+#### Tradeoffs
+
+Pro: Clarifies the meaning of status.
```

Comment:
I see this as a big con, "clarifying" it into something that makes a large number of our resources wrong.